`sendmmsg` in `linux.zig` incorrectly assumes generic `mmsghdr`

### Zig Version

0.10.1 (code is still the same on master however)

### Steps to Reproduce and Observed Behavior

Following minimal reproduction:

```
pub fn main() !void {
    const socket = try os.socket(os.AF.INET, os.SOCK.DGRAM, 0);
    const address = try std.net.Address.resolveIp("127.0.0.1", 20000);
    try os.connect(socket, &address.any, address.getOsSockLen());
    defer os.close(socket);

    const message: [1460]u8 = undefined;
    while (true) {
        var messages: [1024]os.linux.mmsghdr_const = undefined;
        var io_vects: [1024]os.iovec_const = undefined;
        var i: usize = 0;
        while (i < messages.len) : (i += 1) {
            io_vects[i].iov_base = &message;
            io_vects[i].iov_len = message.len;
            messages[i].msg_hdr.iov = @ptrCast([*]os.iovec_const, &io_vects[i]);
            messages[i].msg_hdr.iovlen = 1;
        }

        _ = os.linux.sendmmsg(socket, &messages, messages.len, 0);
    }
}
```

And error:

```
0.10.1/lib/zig/std/os/linux.zig:1243:55: error: type 'type' not a function
    if (@typeInfo(usize).Int.bits > @typeInfo(@TypeOf(mmsghdr(undefined).msg_len)).Int.bits) {
                                                      ^~~~~~~
```

It would appear from `mmsghdr(undefined)` that at some point it was defined as a generic and no longer is, but the `sendmmsg` function was not updated.

- [Location of the error](https://github.com/ziglang/zig/blob/master/lib/std/os/linux.zig#L1271)
- [Definition of mmsghdr](https://github.com/ziglang/zig/blob/master/lib/std/os/linux.zig#L3442)

### Expected Behavior

To be able to call the function.

--- a/lib/std/os/linux.zig
+++ b/lib/std/os/linux.zig
@@ -1268,7 +1268,7 @@ pub fn sendmsg(fd: i32, msg: *const msghdr_const, flags: u32) usize {
 }
 
 pub fn sendmmsg(fd: i32, msgvec: [*]mmsghdr_const, vlen: u32, flags: u32) usize {
-    if (@typeInfo(usize).Int.bits > @typeInfo(@TypeOf(mmsghdr(undefined).msg_len)).Int.bits) {
+    if (@typeInfo(usize).Int.bits > @typeInfo(@typeInfo(mmsghdr).Struct.fields[1].type).Int.bits) {
         // workaround kernel brokenness:
         // if adding up all iov_len overflows a i32 then split into multiple calls
         // see https://www.openwall.com/lists/musl/2014/06/07/5

